python/python/Ice/Py3/IceFuture.py

```diff
@@ -19,25 +20,31 @@ def __await__(self):
             yield self
         return self.result()
 
+
 def wrap_future(future, *, loop=None):
     '''Wrap Ice.Future object into an asyncio.Future.'''
     if isinstance(future, asyncio.Future):
         return future
 
     assert isinstance(future, FutureBase), 'Ice.Future is expected, got {!r}'.format(future)
 
-    if loop is None:
-        loop = asyncio.get_event_loop()
-
-    af = asyncio.Future()
-
-    def callback():
-        if future.cancelled():
-            af.cancel()
-        elif future.exception():
-            af.set_exception(future.exception())
-        else:
-            af.set_result(future.result())
-
-    future.add_done_callback(lambda f: loop.call_soon_threadsafe(callback))
-    return af
+    def forwardCompletion(sourceFuture, targetFuture):
+        if not targetFuture.done():
+            if sourceFuture.cancelled():
+                targetFuture.cancel()
+            elif sourceFuture.exception():
+                targetFuture.set_exception(sourceFuture.exception())
+            else:
+                targetFuture.set_result(sourceFuture.result())
+
+    asyncioFuture = asyncio.Future()
```

Comment:
I think it should be `loop.create_future()`, where loop is asyncio.get_runnning_loop / get_event_loop when set to None.

Reply:
Ok fixed, I kept get_event_loop for now since I assume we still want to support 3.5 on Linux.